test/replication-luatest/election_pre_vote_test.lua

```diff
@@ -0,0 +1,114 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local g = t.group('pre-vote')
+
+local REPLICATION_TIMEOUT = 0.1
+local DEATH_TIMEOUT = 5 * REPLICATION_TIMEOUT
+
+--
+-- gh-6654: pre-vote.
+-- The nodes shouldn't start elections when the elections would be either
+-- disruptive (there's an active leader somewhere) or pointless (the node
+-- doesn't have a quorum of peers).
+--
+g.before_all(function()
+    g.cluster = cluster:new({})
+    g.cfg = {
+        replication = {
+            server.build_instance_uri('node1'),
+            server.build_instance_uri('node2'),
+            server.build_instance_uri('node3'),
+        },
+        election_mode = 'candidate',
+        replication_timeout = REPLICATION_TIMEOUT,
+    }
+    g.node1 = g.cluster:build_and_add_server({alias = 'node1', box_cfg = g.cfg})
+    g.node2 = g.cluster:build_and_add_server({alias = 'node2', box_cfg = g.cfg})
+    g.node3 = g.cluster:build_and_add_server({alias = 'node3', box_cfg = g.cfg})
+    g.cluster:start()
+end)
+
+g.before_each(function()
+    g.leader = g.cluster:get_leader()
+    t.assert(g.leader ~= nil, 'Cluster elected a leader')
+    g.follower1 = g.node1 ~= g.leader and g.node1 or g.node2
+    g.follower2 = g.node3 ~= g.leader and g.node3 or g.node2
+end)
+
+g.after_each(function()
+    g.leader = nil
+    g.follower1 = nil
+    g.follower2 = nil
+end)
+
+--
+-- Test that the node doesn't start elections in the following conditions:
+--  - the cluster has a leader
+--  - the node doesn't see the leader directly
+--  - the node has a quorum of connections to remote peers (excluding leader)
+--
+g.test_no_direct_connection = function(g)
+    local term = g.follower1:exec(function() return box.info.election.term end)
+    g.follower1:exec(function(cfg) box.cfg(cfg) end,
+                     {{
+                         replication = {
+                            server.build_instance_uri(g.follower1.alias),
+                            server.build_instance_uri(g.follower2.alias),
+                         }
+                     }})
```

Comment:
You probably don't need so huge indentation. One tab would be enough. In the other places too.

Reply:
Well, it's the second argument of `exec`, so it was aligned accordingly, but I don't mind. Fixed.

You might have the same question about the new `unit/raft` test: the linter wouldn't let me align it like you suggest. So there are a couple of hugely indented blocks there.